ui/component/or-configuration/src/or-conf-realm/or-conf-realm-card.ts

```diff
@@ -0,0 +1,351 @@
+  }
+
+  protected _setColor(key:string, value:string){
+    const colors  = this._getColors()
+    colors[key] = value
+    let css = ":host > * {"
+    Object.entries(colors).map(([key, value]) => {
+      css += key +":" +value + ";"
+    })
+    this.realm.styles = css
+  }
+
+  protected _setHeader(keys:ManagerHeaders[], list: ManagerHeaders[]){
+    if (!this.realm.headers){
+      this.realm.headers = this.headerListSecondary.concat(this.headerListPrimary)
+    }
+    this.realm.headers = this.realm.headers?.filter(function(ele){
+      return !list.includes(ele);
+    });
+    this.realm.headers = this.realm.headers?.concat(keys)
+  }
+
+  protected _getImagePath(file:File, fileName: string){
+    let extension = "";
+    switch (file.type){
+      case "image/png":
+        extension = "png"
+        break;
+      case "image/jpg":
+        extension = "jpg"
+        break;
+      case "image/jpeg":
+        extension = "jpeg"
+        break;
+      case "image/vnd.microsoft.icon":
+        extension = "ico"
+        break;
+      case "image/svg+xml":
+        extension = "svg"
+        break;
+    }
+    return "/images/" + this.name + "/" + fileName + "." +  extension
+  }
+
+  protected files: {[name:string] : FileInfo} = {}
+
+  protected async _setImageForUpload(file: File, fileName: string) {
+    const path = this._getImagePath(file, fileName)
+    this.files[path] = {
+      // name: 'filename',
+      contents: await this.convertBase64(file),
+      // binary: true
+    } as FileInfo;
+    return path;
+  }
+
+  convertBase64 (file:any) {
+    return new Promise((resolve, reject) => {
+      const fileReader = new FileReader();
+      fileReader.readAsDataURL(file);
+
+      fileReader.onload = () => {
+        resolve(fileReader.result);
+      };
+
+      fileReader.onerror = (error) => {
+        reject(error);
+      };
+    });
+  };
+
+
+  protected _showRemoveRealmDialog(){
+
+    const dialogActions: DialogAction[] = [
+      {
+        actionName: "cancel",
+        content: i18next.t("cancel")
+      },
+      {
+        default: true,
+        actionName: "ok",
+        content: i18next.t("yes"),
+        action: () => {this.onRemove()}},
+
+    ];
+    const dialog = showDialog(new OrMwcDialog()
+      .setHeading(i18next.t('delete'))
+      .setActions(dialogActions)
+      .setContent(html `
+        ${i18next.t('configuration.deleteRealmCustomizationConfirm')}
+      `)
+      .setStyles(html`
+                        <style>
+                            .mdc-dialog__surface {
+                              padding: 4px 8px;
+                            }
+                            #dialog-content {
+                              padding: 24px;
+                            }
+                        </style>
+                    `)
+      .setDismissAction(null));
+
+  }
+
+
+  render() {
+    const colors = this._getColors();
+    const app = this;
+    document.addEventListener('saveManagerConfig', () => {
```

Comment:
This will endlessly create eventlisteners, making multiple active at the same time. Make sure only 1 exists on `firstUpdated()` and then saved onto state, or what ever you solution you find suitable ;)